core/src/main/scala/com/lightbend/paradox/template/PageTemplateSystem.scala

```diff
@@ -23,7 +23,7 @@ import org.stringtemplate.v4.{ STErrorListener, STRawGroupDir, ST }
 import collection.concurrent.TrieMap
 
 object CachedTemplates {
-  def apply(dir: File, templateName: String): PageTemplate = {
+  def apply(dir: File, templateName: String = PageTemplate.DefaultName): PageTemplate = {
```

Comment:
This looks risky. You could call `CachedTemplates(new File("template1"))` and then `CachedTemplates(new File("template2"))` and that second call will give you the template for "template1", because you didn't specify a "templateName".

As an aside, why did we introduce the global mutable state of CachedTemplates?